hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieClient.java

```diff
@@ -262,6 +282,18 @@ protected void finalizeWrite(HoodieTable table, String instantTime, List<HoodieW
     }
   }
 
+  class GetMetadataWriterFunc implements Functions.Function2<String, HoodieTableMetaClient, Option<HoodieTableMetadataWriter>> {
+
+    @Override
+    public Option<HoodieTableMetadataWriter> apply(String triggeringInstantTimestamp, HoodieTableMetaClient metaClient) {
+      return getMetadataWriter(triggeringInstantTimestamp, metaClient);
+    }
+  }
+
+  Option<HoodieTableMetadataWriter> getMetadataWriter(String triggeringInstantTimestamp, HoodieTableMetaClient metaClient) {
+    throw new HoodieException("Each engine's write client is expected to implement this method");
```

Comment:
You can just make this `protected abstract` to require this right?

Reply:
sure.